frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -448,11 +455,15 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
           let order = 1;
           experimentDesignFormData.conditions = experimentDesignFormData.conditions.map(
             (condition, index) => {
+              if (isNaN(condition.assignmentWeight) && condition.assignmentWeight.endsWith("%")) {
+                condition.assignmentWeight = Number(condition.assignmentWeight.slice(0,-1));
+              }
               return this.experimentInfo
                 ? ({ ...this.experimentInfo.conditions[index], ...condition, order: order++ })
                 : ({ id: uuid.v4(), ...condition, name: '', order: order++ });
             }
           );
+          console.log(experimentDesignFormData.conditions);
```

Comment:
oops console log snuck in

Reply:
Hey @danoswaltCL thanks for looking in depth in this PR. Actually I was still working on various fixes and few commits are yet to be pushed as I wanted to create a draft PR. Still a lot of your detailed feedback would help me a lot. Thanks again. 

Clearing a few things:
1. I just pushed a commit that will allow an empty metric list in stepper to create an experiment and later add them in edit mode.
2. Removed the extra console log in my last commit
3. The floating "equal" checkbox is yet to be improved. I had looked into the figma design, but placing it in header cell below the titles for the mat-table was not working, still have an idea to try again. Just kept it above the title for now.
4.  Disabled "required" as it wont be used as of now. We need to implement this in coordinated site/target.
5. In Edit mode "Grouped and Single" metric keys are kept as disabled and user needs to delete the complete metric if they want to edit it and make it group to single or vice-versa as now we don't have explicit sections for two different metric types and it became very complex to handle change of type in edit mode. I have to comment this in figma design. Rest fields are editable.

Rest things I will make changes accordingly.